src/figuresClasses.ts

```diff
@@ -1,11 +1,75 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: 'triangle' = 'triangle',
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('your error message');
```

Comment:
The error message 'your error message' is a placeholder. Consider providing a more descriptive error message, such as 'Side lengths must be positive numbers.'